inventory-consumer/handler.go

```diff
@@ -37,15 +37,27 @@ func (this *handler) onMessage(ctx context.Context, msg kafka.Message) {
 				log.Println("Error retrieving state for account: ", value.Host.Account)
 			}
 
-			client := []domain.Host{value.Host}
-
-			// TODO: Switch on event type. Once config-manager is updating rhc_config_state in inventory
-			// a check can be made on the rhc_config_state id to determine if work should be done.
-			responses, err := this.ConfigManagerService.ApplyState(ctx, accState, client)
+			reqID, err := kafkaUtils.GetHeader(msg, "request_id")
 			if err != nil {
-				log.Println("Error applying state: ", err)
+				log.Println("Error getting request_id: ", err)
+				return
```

Comment:
I wonder if instead of bailing out here, if you should just log that there wasn't a request_id...generate a new one, log it and carry on processing the host?

Reply:
Agreed!